coderd/rbac/policy.rego

```diff
@@ -2,392 +2,424 @@ package authz
 
-default scope_user := 0
-scope_user := user_allow([input.subject.scope])
+default org_member := 0
 
-user_allow(roles) := num if {
+org_member := num if {
+	# Object must be jointly owned by the user
 	input.object.owner != ""
 	input.subject.id = input.object.owner
-
-	allow := {is_allowed |
-		# Iterate over all user permissions in all roles
-		perm := roles[_].user[_]
-		perm.action in [input.action, "*"]
-		perm.resource_type in [input.object.type, "*"]
-
-		# is_allowed is either 'true' or 'false' if a matching permission exists.
-		is_allowed := bool_flip(perm.negate)
-	}
-	num := number(allow)
+	num := check_org_permissions(input.subject.roles, "member")
 }
 
-# Scope allow_list is a list of resource (Type, ID) tuples explicitly allowed by the scope.
-# If the list contains `(*,*)`, then all resources are allowed.
-scope_allow_list if {
-	input.subject.scope.allow_list[_] == {"type": "*", "id": "*"}
-}
+default scope_org_member := 0
 
-# This is a shortcut if the allow_list contains (type, *), then allow all IDs of that type.
-scope_allow_list if {
-	input.subject.scope.allow_list[_] == {"type": input.object.type, "id": "*"}
+scope_org_member := num if {
+	# Object must be jointly owned by the user
+	input.object.owner != ""
+	input.subject.id = input.object.owner
+	num := check_org_permissions([input.subject.scope], "member")
 }
 
-# A comprehension that iterates over the allow_list and checks if the
-# (object.type, object.id) is in the allowed ids.
-scope_allow_list if {
-	# If the wildcard is listed in the allow_list, we do not care about the
-	# object.id. This line is included to prevent partial compilations from
-	# ever needing to include the object.id.
-	not {"type": "*", "id": "*"} in input.subject.scope.allow_list
-	# This is equivalent to the above line, as `type` is known at partial query time.
-	not {"type": input.object.type, "id": "*"} in input.subject.scope.allow_list
+#==============================================================================#
+# Role rules                                                                   #
+#==============================================================================#
 
-	# allows_ids is the set of all ids allowed for the given object.type
-	allowed_ids := {allowed_id |
-  		# Iterate over all allow list elements
-  		ele := input.subject.scope.allow_list[_]
-  		ele.type in [input.object.type, "*"]
-      allowed_id := ele.id
-  }
-
-  # Return if the object.id is in the allowed ids
-  # This rule is evaluated at the end so the partial query can use the object.id
-  # against this precomputed set of allowed ids.
-  input.object.id in allowed_ids
-}
-
-# -------------------
-# Role-Specific Rules
-# -------------------
+# role_allow specifies all of the conditions under which a role can grant
+# permission
 
+# Site level authorization
 role_allow if {
 	site = 1
 }
 
+# User level authorization
+role_allow if {
+	not site = -1
+
+	user = 1
+}
```

Comment:
We had `org_ok` here before, which essentially checked `input.object.org_owner == ""`

We need to put that back, but we can probably just do `input.object.org_owner == ""`

Same comment for the scope too